crates/red_knot_python_semantic/resources/mdtest/subscript/tuple.md

```diff
@@ -117,6 +117,6 @@ from typing import Tuple
 
 class C(Tuple): ...
 
-# revealed: tuple[Literal[C], Literal[tuple], Literal[Sequence], Literal[Reversible], Literal[Collection], Literal[Iterable], Literal[Container], @Todo(protocol), Literal[object]]
+# revealed: tuple[Literal[C], Literal[tuple], Literal[Sequence], Literal[Reversible], Literal[Collection], Literal[Iterable], Literal[Container], @Todo(`Protocol[]` subscript), Literal[object]]
```

Comment:
I guess this `Todo(Protocol subscript)` should turn into `Generic` in the MRO? At runtime this is 

```
[<class '__main__.C'>, <class 'tuple'>, <class 'typing.Generic'>, <class 'object'>]
```

Generic is there but Protocol is not. I'm not really sure why anything related to `Protocol` should show up here even now?

Reply:
The MROs of builtin containers are all very different in typeshed to what they are at runtime. I'm not sure it makes sense for red-knot to distinguish between `tuple` and `Tuple`, and at runtime `tuple` obviously inherits directly from `object`. But in typeshed, `tuple` inherits from `typing.Sequence`, which inherits from `Reversible` and `Collection`. `Collection` and `Reversible` are both protocol classes, which is how `Protocol` ends up in the MRO here.

I can see if I can improve this to include `Generic` in the MRO as well without breaking the special-casing that means we don't emit false positives when users subscript classes that inherit from `Protocol[T]` or `Generic[T]`

Reply:
On the latest version of the PR, this is now

```
revealed: tuple[Literal[C], Literal[tuple], Literal[Sequence], Literal[Reversible], Literal[Collection], Literal[Iterable], Literal[Container], @Todo(`Protocol[]` subscript), @Todo(`Generic[]` subscript), Literal[object]]
```

Reply:
no worries -- it actually simplified some special-casing elsewhere, so I think it was a worthwhile change!